src/socket_base.cpp

```diff
@@ -1414,6 +1414,7 @@ void zmq::socket_base_t::update_pipe_options(int option_)
         for (pipes_t::size_type i = 0; i != pipes.size(); ++i)
         {
             pipes[i]->set_hwms(options.rcvhwm, options.sndhwm);
+            pipes[i]->set_peer_hwms(options.sndhwm, options.rcvhwm);
```

Comment:
I don't think this is safe. In the inproc case, this might be owned and used from another thread. @ricnewton any opinion?